traffic_incident_parser/src/traffic_incident_parser_worker.cpp

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright (C) 2020 LEIDOS.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+#include "traffic_incident_parser_worker.h"
+
+namespace traffic
+{
+
+  TrafficIncidentParserWorker::TrafficIncidentParserWorker(carma_wm::WorldModelConstPtr wm,PublishTrafficControlCallback traffic_control_pub) : traffic_control_pub_(traffic_control_pub),wm_(wm){};
+
+  void TrafficIncidentParserWorker::mobilityOperationCallback(const cav_msgs::MobilityOperation &mobility_msg)
+  {
+   
+        if(mobility_msg.strategy=="carma3/Incident_Use_Case")
+      { 
+        if(mobility_msg.strategy_params!=previous_strategy_params)
+        {  
+           previous_strategy_params=mobility_msg.strategy_params;
+           mobilityMessageParser(mobility_msg.strategy_params);
+           traffic_control_pub_(composeTrafficControlMesssage());
+        }
+      }
+   }
+
+  void TrafficIncidentParserWorker::projectionCallback(const std_msgs::String &projection_msg)
+  {
+      projection_msg_=projection_msg.data;
+   }
+
+
+  void TrafficIncidentParserWorker::mobilityMessageParser(std::string mobility_strategy_params)
+  {
+
+   std::vector<std::string> vec={};
+   std::string delimiter = ",";
+   size_t pos = 0;
+   std::string token;
+    while ((pos = mobility_strategy_params.find(delimiter)) != std::string::npos) 
+    {
+        token = mobility_strategy_params.substr(0, pos);
+        vec.push_back(token);
+        mobility_strategy_params.erase(0, pos + delimiter.length());
+    }
+    vec.push_back(mobility_strategy_params);
+
```

Comment:
Let's put a safeguard here like
`if (vec.size() != 5)
ROS_ERROR_STREAM("Given mobility strategy params are not correctly formatted.")
return;`
so we don't run into segfault from the later lines